src/request.ts

```diff
@@ -231,6 +236,20 @@ class Request {
     const cloudUrl: string = Auth.getEndpointForResource(hostname, cloudType);
     options.url = options.url!.replace(hostname, cloudUrl);
   }
+
+  private createProxyConfigFromString(url: string): AxiosProxyConfig {
+    const parsedUrl = new URL(url);
+    const hostname = parsedUrl.hostname;
```

Comment:
Since we're using hostname in just one place, let's not create a separate const for it and instead assign it directly in the returned object.